src/retriable.rs

```diff
@@ -0,0 +1,32 @@
+#[macro_export]
+macro_rules! execute_retriable_or_panic {
```

Comment:
I feel it is generic enough to drop "execute". Maybe `await_and_retry_or_panic`? Ugly names for macro solutions :smile:  Please, think of a better name :pray: